### DISCUSSION TITLE
How to run all models at once?

### DISCUSSION THREAD

**Discussion by rullator** [ENHANCE-PET/MOOSE#71] [Q&A] [answered]
Hi all,

as stated MOOSE can simply be started by typing:
`moosez -d <path_to_image_dir> -m clin_ct_organs`

Im comparison to the earlier version, seperated models exist. But how can I start MOOSE aiming to run all models at once?
I already tried:
`moosez -d <path_to_image_dir> -m clin_ct_organs -m clin_ct_cardiac`
-> It uses only the last given model.

`moosez -d <path_to_image_dir> -m clin_ct_organs clin_ct_cardiac`
-> Returns an error: `moosez: error: unrecognized arguments: clin_ct_cardiac`

Running sequentially
`moosez -d <path_to_image_dir> -m clin_ct_organs `
`moosez -d <path_to_image_dir> -m clin_ct_cardiac`
-> Runs into error since it creates a folder within each subject (`moosez-clin_*`) and MOOSE cannot handle the created folder in the second run (`IsADirectoryError: [Errno 21] Is a directory: 'SUBJECT/moosez-clin_ct_organs-2023-10-16-12-14-09/CT'`).

Can someone possibly help?

PS: Wasn't sure whether this question belongs to "Issues" or whether I'm just on the wrong way and missed something..

**Comment by LalithShiyam** [comment]
Hi @rullator, many thanks for asking this question! I think this is a feature request, you are right, you need to clean up the generated `moosez` folder before running the next model. I realize that it can be annoying. I think I can make a quick patch to ignore these `moosez` generated folders. And you can run it one by one (sequentially). 

E.g: `moosez -d path_to_dir -m your_model_1`
And 
`moosez -d path_to_dir -m your_model_2`

Would that be a reasonable fix for you?

**Comment by rullator** [comment]
That would help a lot :)
Thanks!

**Reply by LalithShiyam** [reply]
Hi @rullator: Fixed this in the latest [commit](https://github.com/QIMP-Team/MOOSE/commit/13981b22f79c84a5b1c69c66e8f9541a280adeeb)!

**Comment by LalithShiyam** [comment, marked as answer]
@rullator : `moosez 2.2.36` is your friend! A patch was included to ignore the previously generated `moosez` folder. Let me know if it works.

`pip install --upgrade moosez`
